src/components/Header/NavDropdown/index.tsx

```diff
@@ -30,17 +30,26 @@ const NavDropdownItem: React.VFC<NavDropdownItemType> = ({
   href,
   text,
   to,
-}) => (
-  <MenuItem
-    as={to && Link}
-    to={to}
-    href={href}
-    data-cy={itemDataCy}
-    onClick={closeMenu}
-  >
-    {text}
-  </MenuItem>
-);
+}) => {
+  const isLink = to !== undefined;
```

Comment:
[nit] This component functions as a link in both cases, so maybe naming this `isReactRouterLink` or `isInternalLink` would be more helpful.